llvm/lib/Target/LoongArch/LoongArchInstrInfo.td

```diff
@@ -713,7 +713,9 @@ class STORE_2RI14<bits<32> op>
 let hasSideEffects = 0, mayLoad = 1, mayStore = 1, Constraints = "@earlyclobber $rd" in
 class AM_3R<bits<32> op>
     : Fmt3R<op, (outs GPR:$rd), (ins GPR:$rk, GPRMemAtomic:$rj),
```

Comment:
AM_3R is incompatible with amcas,
Unlike other am** instructions, amcas rd rk rj , rd should be read, pervious version consider rd write only.

Reply:
> AM_3R is incompatible with amcas, Unlike other am** instructions, amcas rd rk rj , rd should be read, pervious version consider rd write only.

Thanks for the suggestion, I have posted #114508 to precede this PR.